x-pack/qa/src/main/java/org/elasticsearch/xpack/test/rest/IndexMappingTemplateAsserter.java

```diff
@@ -51,9 +51,10 @@
      * effect of a different test running in the cluster.
      *
      * @param client The rest client
+     * @param allowSystemIndexWarnings Whether deprecation warnings for system index access should be allowed/expected.
      * @throws IOException On error
      */
-    public static void assertMlMappingsMatchTemplates(RestClient client) throws IOException {
+    public static void assertMlMappingsMatchTemplates(RestClient client, boolean allowSystemIndexWarnings) throws IOException {
```

Comment:
I can't see this new parameter being used. Am I missing something?

Reply:
Good catch, I forgot some plumbing here. I'll fix - if this is `false`, it should then pass `false` to all of the `assertLegacyTemplateMatchesIndexMappings` calls (as opposed to now, where some of them are hardcoded to `true`)

Reply:
Actually, it turns out I misread one of the callsites and this must be true for both places this is called, so I removed this parameter and went back to specifying the values passed to `assertLegacyTemplateMatchesIndexMappings` directly.